File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        HashMap<Character,Integer> map = new HashMap<Character,Integer>();

        int i = 0, j = 0, ans = 0;

        while (j < s.length()) {
            map.put(s.charAt(j), map.getOrDefault(s.charAt(j), 0) + 1);
            while (map.get(s.charAt(j)) > 2) {
                map.put(s.charAt(i), map.get(s.charAt(i)) - 1);
                i++;
            }
            ans = Math.max(ans, j - i + 1);
            j++;
        }
        return ans;
    }
}

File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cs
public class Solution {
    public int MaximumLengthSubstring(string s) {
        Dictionary<char,int> map =new Dictionary<char,int>();
        int i =0 , j = 0 , ans = 0;
        while(j<s.Length){
            if(map.ContainsKey(s[j]))
            map[s[j]]++;
            else
                map[s[j]]=1;
            while(map[s[j]]>2){
                map[s[i]]--;
                i++;
            }
            ans = Math.Max(ans,j-i+1);
            j++;

        }
        return ans;
        
    }
}